frontstage/views/account/account.py

```diff
@@ -1,20 +1,76 @@
 import logging
 
-from flask import render_template, request
+from flask import render_template, request, flash, url_for
 from structlog import wrap_logger
+from werkzeug.utils import redirect
 
 from frontstage.common.authorisation import jwt_authorization
 from frontstage.controllers import party_controller
+from frontstage.exceptions.exceptions import ApiError
+from frontstage.models import OptionsForm, ContactDetailsChangeForm
 
 from frontstage.views.account import account_bp
 
 logger = wrap_logger(logging.getLogger(__name__))
 
 
-@account_bp.route('/', methods=['GET'])
+@account_bp.route('/', methods=['GET', 'POST'])
 @jwt_authorization(request)
 def get_account(session):
+    form = OptionsForm()
+    form_valid = form.validate()
     party_id = session.get_party_id()
     respondent_details = party_controller.get_respondent_party_by_id(party_id)
-    return render_template('account/account.html', respondent=respondent_details)
+    if request.method == 'POST':
+        if not form_valid:
+            flash('At least one option should be selected!')
+            return redirect(url_for('account_bp.get_account'))
+        if form.data['option'] == 'contact_details':
+            return redirect(url_for('account_bp.change_account_details'))
+    else:
+        return render_template('account/account.html', form=form, respondent=respondent_details)
 
+
+@account_bp.route('/change-account-details', methods=['GET', 'POST'])
+@jwt_authorization(request)
+def change_account_details(session):
+    form = ContactDetailsChangeForm(request.values)
+    party_id = session.get_party_id()
+    respondent_details = party_controller.get_respondent_party_by_id(party_id)
+    update_required_flag = False
+    if request.method == 'POST' and form.validate():
+        logger.info('Attempting to create account')
```

Comment:
This needs changing, we're changing the details not creating a new one

Reply:
👍